internal/kafka/internal/config/dynamic_scaling_config.go

```diff
@@ -52,14 +52,19 @@ func (c *DynamicScalingConfig) validate() error {
 }
 
 type InstanceTypeDynamicScalingConfig struct {
-	ComputeNodesConfig *DynamicScalingComputeNodesConfig `yaml:"compute_nodes_config"`
+	ComputeNodesConfig  *DynamicScalingComputeNodesConfig `yaml:"compute_nodes_config"`
+	CapacityReservation int                               `yaml:"capacity_reservation"`
 }
 
 func (c *InstanceTypeDynamicScalingConfig) validate() error {
 	if c.ComputeNodesConfig == nil {
 		return fmt.Errorf("compute_nodes_config is mandatory")
 	}
 
+	if c.CapacityReservation <= 0 {
```

Comment:
I am wondering if we should allow for this to be set to `0` to express the `no capacity reservation` intent.
What do you think? @JasonPowr @JameelB @VanillaSpoon

Reply:
I am leaning toward making it optional i.e not mandatory. If it is set to `0` or left out (thus defaults to `0`) we'd consider it as no reservation intent. 

Just to avoid making it a nullable field and to have to handle `config !=nil` where it'll be used.